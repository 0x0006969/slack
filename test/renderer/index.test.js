const { Issue, PullRequest } = require('./../../lib/renderer');
const issuesOpened = require('./../fixtures/webhooks/issues.opened.json');
const issuesClosed = require('./../fixtures/webhooks/issues.closed.json');
const combinedStatus = require('./../fixtures/combined_status.json');

const pullRequestOpened = require('./../fixtures/webhooks/pull_request.opened.json');
const pullRequestClosed = require('./../fixtures/webhooks/pull_request.closed.json');

// once everything extends Message,
// we should `expect` there to be a fallback
describe('Issue rendering', () => {
  test('works for unfurls', async () => {
    const issueMessage = new Issue(
      issuesOpened.issue,
      issuesOpened.repository,
      'issues.opened',
      true,
    );
    const rendered = issueMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });

  test('works for notification messages', async () => {
    const issueMessage = new Issue(
      issuesOpened.issue,
      issuesOpened.repository,
      'issues.opened',
      false,
    );
    const rendered = issueMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
  test('works for minor notifications', async () => {
    const issueMessage = new Issue(
      issuesClosed.issue,
      issuesClosed.repository,
      'issues.closed',
      false,
    );
    const rendered = issueMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
});

describe('Pull request rendering', () => {
  test('works for notifcation messages', async () => {
    const pullRequest = {
      ...pullRequestOpened.pull_request,
      labels: [],
    };
    const prMessage = new PullRequest(
      pullRequest,
      pullRequestOpened.repository,
      'pull_request.opened',
      false,
      null,
    );
    const rendered = prMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });

  test('works for notifcation messages with statuses', async () => {
    const pullRequest = {
      ...pullRequestOpened.pull_request,
      labels: [],
    };
    const prMessage = new PullRequest(
      pullRequest,
      pullRequestOpened.repository,
      'pull_request.opened',
      false,
      combinedStatus.statuses,
    );
    const rendered = prMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });

  test('works for minor notifications', async () => {
    const pullRequest = {
      ...pullRequestClosed.pull_request,
      labels: [],
    };
    const prMessage = new PullRequest(
      pullRequest,
      pullRequestOpened.repository,
      'pull_request.closed',
      false,
      null,
    );
    const rendered = prMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });

  test('works for unfurls', async () => {
    const pullRequest = {
      ...pullRequestOpened.pull_request,
      labels: [],
    };
    const prMessage = new PullRequest(
      pullRequest,
      pullRequestOpened.repository,
      'pull_request.opened',
      true,
      null,
    );
    const rendered = prMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
});
