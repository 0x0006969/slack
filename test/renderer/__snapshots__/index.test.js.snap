// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue rendering works for minor notifications 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#cb2431",
      "fallback": "[github-slack/public-test] Issue closed by wilhelmklopp",
      "footer": "<https://github.com/github-slack/public-test/issues/1|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "pretext": "[github-slack/public-test] Issue closed by wilhelmklopp",
      "title": "#1 Needs content",
      "title_link": "https://github.com/github-slack/public-test/issues/1",
      "ts": 1508171038,
    },
  ],
}
`;

exports[`Issue rendering works for notification messages 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#36a64f",
      "fallback": "[github-slack/public-test] Issue opened by wilhelmklopp",
      "fields": Array [
        Object {
          "short": true,
          "title": "Assignees",
          "value": "wilhelmklopp",
        },
        Object {
          "short": true,
          "title": "Labels",
          "value": "enhancement, question, wontfix",
        },
      ],
      "footer": "<https://github.com/github-slack/public-test/issues/1|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "pretext": "[github-slack/public-test] Issue opened by wilhelmklopp",
      "text": "This repo needs some content, as currently, it does not have **any**.",
      "title": "#1 Needs content",
      "title_link": "https://github.com/github-slack/public-test/issues/1",
      "ts": 1508171038,
    },
  ],
}
`;

exports[`Issue rendering works for unfurls 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#36a64f",
      "fallback": "#1 Needs content",
      "fields": Array [
        Object {
          "short": true,
          "title": "Assignees",
          "value": "wilhelmklopp",
        },
        Object {
          "short": true,
          "title": "Labels",
          "value": "enhancement, question, wontfix",
        },
      ],
      "footer": "<https://github.com/github-slack/public-test/issues/1|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "This repo needs some content, as currently, it does not have **any**.",
      "title": "#1 Needs content",
      "title_link": "https://github.com/github-slack/public-test/issues/1",
      "ts": 1508171038,
    },
  ],
}
`;

exports[`Pull request rendering works for minor notifications 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars0.githubusercontent.com/u/173?v=4",
      "author_link": "https://github.com/bkeepers",
      "author_name": "bkeepers",
      "color": "#6f42c1",
      "fallback": "#36 Rearrange logic",
      "footer": "<https://github.com/github-slack/app/pull/36|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "title": "#36 Rearrange logic",
      "title_link": "https://github.com/github-slack/app/pull/36",
      "ts": 1506619320,
    },
  ],
}
`;

exports[`Pull request rendering works for notifcation messages 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#36a64f",
      "fallback": "[github-slack/app] Pull request opened by wilhelmklopp",
      "mrkdwn_in": Array [
        "text",
      ],
      "pretext": "[github-slack/app] Pull request opened by wilhelmklopp",
      "text": "1 commit into \`github-slack:master\` from \`github-slack:wilhelmklopp/add-readme-badges\`",
    },
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#36a64f",
      "fallback": "#31 Add badges to readme",
      "fields": Array [],
      "footer": "<https://github.com/github-slack/app/pull/31|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "![image](https://user-images.githubusercontent.com/7718702/30750617-39e2fb52-9fb7-11e7-8da1-d6542317bee2.png)
",
      "title": "#31 Add badges to readme",
      "title_link": "https://github.com/github-slack/app/pull/31",
      "ts": 1506092298,
    },
  ],
}
`;

exports[`Pull request rendering works for notifcation messages with statuses 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#36a64f",
      "fallback": "[github-slack/app] Pull request opened by wilhelmklopp",
      "mrkdwn_in": Array [
        "text",
      ],
      "pretext": "[github-slack/app] Pull request opened by wilhelmklopp",
      "text": "1 commit into \`github-slack:master\` from \`github-slack:wilhelmklopp/add-readme-badges\`",
    },
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#36a64f",
      "fallback": "#31 Add badges to readme",
      "fields": Array [],
      "footer": "<https://github.com/github-slack/app/pull/31|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "![image](https://user-images.githubusercontent.com/7718702/30750617-39e2fb52-9fb7-11e7-8da1-d6542317bee2.png)
",
      "title": "#31 Add badges to readme",
      "title_link": "https://github.com/github-slack/app/pull/31",
      "ts": 1506092298,
    },
    Object {
      "author_icon": "https://avatars2.githubusercontent.com/oa/1508?v=4",
      "author_link": "https://travis-ci.org/github-slack/app/builds/278661171?utm_source=github_status&utm_medium=notification",
      "author_name": "continuous-integration/travis-ci/pr",
      "color": "#dbab09",
      "fallback": "The Travis CI build is pending",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "The Travis CI build is pending",
    },
    Object {
      "author_icon": "https://avatars2.githubusercontent.com/oa/1508?v=4",
      "author_link": "https://travis-ci.org/github-slack/app/builds/278661144?utm_source=github_status&utm_medium=notification",
      "author_name": "continuous-integration/travis-ci/push",
      "color": "#28a745",
      "fallback": "The Travis CI build passed",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "The Travis CI build passed",
    },
    Object {
      "author_icon": "https://avatars2.githubusercontent.com/in/254?v=4",
      "author_link": "https://codecov.io/gh/github-slack/app/compare/0143c3cd70052e00bb07dc09229b261b3afd807d...68c2ef08b77a144aeaa69b601ecbacb979e3a8fd",
      "author_name": "codecov/patch",
      "color": "#cb2431",
      "fallback": "Coverage not affected when comparing 0143c3c...68c2ef0",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Coverage not affected when comparing 0143c3c...68c2ef0",
    },
    Object {
      "author_icon": "https://avatars2.githubusercontent.com/in/254?v=4",
      "author_link": "https://codecov.io/gh/github-slack/app/compare/0143c3cd70052e00bb07dc09229b261b3afd807d...68c2ef08b77a144aeaa69b601ecbacb979e3a8fd",
      "author_name": "codecov/project",
      "color": "#cb2431",
      "fallback": "98.37% remains the same compared to 0143c3c",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "98.37% remains the same compared to 0143c3c",
    },
  ],
}
`;

exports[`Pull request rendering works for unfurls 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars1.githubusercontent.com/u/7718702?v=4",
      "author_link": "https://github.com/wilhelmklopp",
      "author_name": "wilhelmklopp",
      "color": "#36a64f",
      "fallback": "#31 Add badges to readme",
      "fields": Array [],
      "footer": "<https://github.com/github-slack/app/pull/31|View it on GitHub>",
      "footer_icon": "https://assets-cdn.github.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "![image](https://user-images.githubusercontent.com/7718702/30750617-39e2fb52-9fb7-11e7-8da1-d6542317bee2.png)
",
      "title": "#31 Add badges to readme",
      "title_link": "https://github.com/github-slack/app/pull/31",
      "ts": 1506092298,
    },
  ],
}
`;
